pkg/kotsadmsnapshot/backup.go

```diff
@@ -191,254 +201,512 @@ func CreateApplicationBackup(ctx context.Context, a *apptypes.App, isScheduled b
+		renderedBackup, err := helper.RenderAppFile(appMeta.app, nil, []byte(backupSpec), appMeta.kotsKinds, metadata.kotsadmNamespace)
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to render backup")
 		}
-		veleroBackup, err := kotsutil.LoadBackupFromContents(renderedBackup)
+		appVeleroBackup, err = kotsutil.LoadBackupFromContents(renderedBackup)
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to load backup from contents")
 		}
 
-		// ** merge app backup info ** //
-		// included namespaces
-		includedNamespaces = append(includedNamespaces, veleroBackup.Spec.IncludedNamespaces...)
-		includedNamespaces = append(includedNamespaces, kotsKinds.KotsApplication.Spec.AdditionalNamespaces...)
+		appVeleroBackup.Name = ""
+		appVeleroBackup.GenerateName = "application-"
 
-		// excluded namespaces
-		excludedNamespaces = append(excludedNamespaces, veleroBackup.Spec.ExcludedNamespaces...)
+		break
+	}
 
-		// annotations
-		for k, v := range veleroBackup.Annotations {
-			backupAnnotations[k] = v
-		}
+	if appVeleroBackup == nil {
+		return nil, nil
+	}
 
-		// ordered resources
-		for k, v := range veleroBackup.Spec.OrderedResources {
-			backupOrderedResources[k] = v
-		}
+	var err error
+	appVeleroBackup.Annotations, err = appendCommonAnnotations(k8sClient, appVeleroBackup.Annotations, metadata, true)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to add annotations to application backup")
+	}
+	appVeleroBackup.Annotations[InstanceBackupTypeAnnotation] = InstanceBackupTypeApp
+
+	appVeleroBackup.Spec.StorageLocation = "default"
 
-		// backup hooks
-		backupHooks.Resources = append(backupHooks.Resources, veleroBackup.Spec.Hooks.Resources...)
+	if metadata.snapshotTTL > 0 {
+		appVeleroBackup.Spec.TTL = metav1.Duration{
+			Duration: metadata.snapshotTTL,
+		}
 	}
 
-	veleroClient, err := veleroclientv1.NewForConfig(cfg)
+	return appVeleroBackup, nil
+}
+
+// mergeAppBackupSpec merges the app backup spec into the velero backup spec when improved DR is
+// disabled. Unsupported fields that are intentionally left out because they might break full
+// snapshots:
+// - includedResources
+// - excludedResources
+// - labelSelector
+func mergeAppBackupSpec(backup *velerov1.Backup, appMeta appInstanceBackupMetadata, kotsadmNamespace string, isEC bool) error {
+	backupSpec, err := appMeta.kotsKinds.Marshal("velero.io", "v1", "Backup")
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to create velero clientset")
+		return errors.Wrap(err, "failed to get backup spec from kotskinds")
 	}
 
-	kotsadmVeleroBackendStorageLocation, err := kotssnapshot.FindBackupStoreLocation(ctx, clientset, veleroClient, kotsadmNamespace)
+	if backupSpec == "" {
+		// If this is Embedded Cluster, backups are always enabled and we must include the
+		// namespace.
+		if isEC {
+			backup.Spec.IncludedNamespaces = append(backup.Spec.IncludedNamespaces, appMeta.kotsKinds.KotsApplication.Spec.AdditionalNamespaces...)
```

Comment:
i think this is missing from the improved DR workflow. i don't see where additional namespaces get added in that case

Reply:
Why would we add additional namespaces in the improved dr workflow. Shouldnt the customer just include them in their spec?

Reply:
oh you're right 🤦